.github/workflows/build-and-test.yml

```diff
@@ -433,3 +433,115 @@ jobs:
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: arm64
+
+      - name: Downloading binaries-linux_${{ matrix.ARCH }}
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries-linux_${{ matrix.ARCH }}
+          path: ./bin
+      
+      - name: Build ${{ matrix.ARCH }} docker image
+        run: |
+          make docker-otelcol SKIP_COMPILE=true ARCH=${{ matrix.ARCH }}
+
+      - name: Run docker image
+        run: |
+          # ensure that the arch in the image manifest is correct
+          [ "$( docker inspect --format='{{.Architecture}}' otelcol:${{ matrix.ARCH }} )" = "${{ matrix.ARCH }}" ] || exit 1
+          docker run --platform linux/${{ matrix.ARCH }} -d -e SPLUNK_ACCESS_TOKEN=12345 -e SPLUNK_REALM=fake-realm --name otelcol otelcol:${{ matrix.ARCH }}
+          sleep 10
+          if [ -z "$( docker ps --filter=status=running --filter=name=otelcol -q )" ]; then
+            docker logs otelcol
+            echo "Failing job execution: fail to start otelcol docker container in 10 seconds."
+            exit 1
+          fi
+      
+      - name: Save docker image
+        run: |
+          mkdir -p dist
+          docker save -o dist/otelcol.tar otelcol:latest
+      
+      - name: Uploading Linux docker image artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: linux-docker-image
+          path: ./dist/otelcol.tar
+
+  docker-otelcol-windows:
+    name: docker-otelcol-windows
+    runs-on: windows-latest
+    needs: [cross-compile]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading binaries-windows_amd64
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries-windows_amd64
+          path: ./bin
+
+      - name: Build docker image
+        run: |
+          $ErrorActionPreference = 'Stop'
+          Copy-Item .\bin\otelcol_windows_amd64.exe .\cmd\otelcol\otelcol.exe
+          Copy-Item .\bin\translatesfx_windows_amd64.exe .\cmd\otelcol\translatesfx.exe
+          docker build -t otelcol-windows --build-arg SMART_AGENT_RELEASE=$((Get-Content internal\buildscripts\packaging\smart-agent-release.txt).TrimStart("v")) -f .\cmd\otelcol\Dockerfile.windows .\cmd\otelcol\
+          Remove-Item .\cmd\otelcol\otelcol.exe
+          Remove-Item .\cmd\otelcol\translatesfx.exe
+      
+      - name: Run docker image
+        run: |
+          $ErrorActionPreference = 'Stop'
+          docker run -d -e SPLUNK_ACCESS_TOKEN=12345 -e SPLUNK_REALM=fake-realm --name otelcol otelcol-windows:latest
+          Start-Sleep 10
+          $DockerOutput=$(docker ps --filter=status=running --filter=name=otelcol -q)
+          if ( $DockerOutput -eq $null )
+          {
+            docker logs otelcol
+            echo "Failing job execution: fail to start otelcol docker container in 10 seconds."
+            exit 1
+          }
+      
+      - name: Save docker image
```

Comment:
Let's skip saving and archiving the image to save time.

Reply:
Removed save docker image step for the GitHub Action workflow now.